tests/test_python_metrics_dependencies.py

```diff
@@ -0,0 +1,66 @@
+'''
+Verifies that the packages used by the Python metrics orchestrator have valid versions
+
+Please contact sohier@ or lipovetz@ if you need to push an update that would cause these tests to fail.
+'''
+
+import subprocess
+import sys
+import unittest
+
+import numpy as np
+import packaging.version
+import pandas as pd
+
+
+class TestLinuxDependencies(unittest.TestCase):
+    '''
+    Tests for the existence of all libraries the orchestrator calls via subprocess
+    These APIs should be quite stable so we won't validate the exact versions
+    '''
+    def test_7z(self):
+        result = subprocess.run('which 7z', shell=True, capture_output=True)
+        assert len(result.stdout) > 0
+
+    def test_gunzip(self):
+        result = subprocess.run('which gunzip', shell=True, capture_output=True)
+        assert len(result.stdout) > 0
+
+    def test_ls(self):
+        result = subprocess.run('which ls', shell=True, capture_output=True)
+        assert len(result.stdout) > 0
+
+    def test_pytest(self):
+        result = subprocess.run('which pytest', shell=True, capture_output=True)
+        assert len(result.stdout) > 0
+
+
+class TestPythonDependencies(unittest.TestCase):
+    '''
+    Tests for the orchestrator's Python dependencies.
+    All standard library imports should be covered by the python 3.10.x check.
+    '''
+    def test_python_version(self):
+        # equivalent to py version == 3.10.x
+        assert sys.version_info.major == 3
+        assert sys.version_info.minor == 10
+
+    def test_pandas_version(self):
+        '''
+        The python metrics team will need advance notice before a migration to Pandas 2.1
+        '''
+        assert packaging.version.parse(pd.__version__) >= packaging.version.parse('1.5')
+        assert packaging.version.parse(pd.__version__) < packaging.version.parse('2.1')
```

Comment:
For version range limits, it might also make sense to actually set those in the Dockerfile as well:
https://github.com/Kaggle/docker-python/blob/main/Dockerfile.tmpl

Out of curiosity, what are the limitations that cause us to need these particular versions?
ex. would it make more sense that the specific code relied upon is tested vs. the version?

Reply:
I expect that we will be able to upgrade these libraries further without actually needing to change anything, but want to make sure that we run tests before doing so. I updated the top level comment to hopefully make that more clear. Since this isn't really a hard pin should I still update the Dockerfile? We

Reply:
Gotcha, as these are they will break our image build if these packages upgrade ever, which isn't a great way to test the image. Is there a more realistic test we can run that verifies their behavior instead?

We also have e2e tests which block image promotion and could run the specific kinds of notebooks you want to make sure don't break as part of those instead, since that wouldn't break out image build and might actually prevent the kinds of breakages you care about.

Reply:
We discussed this offline, and decided to actually run the proper metrics tests as part of the image release gating process instead.